subprojects/core/src/main/java/org/gradle/plugin/use/PluginId.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugin.use;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.Nullable;
+
+/**
+ * A description of a plugin.
+ *
+ * @since 3.5
+ */
+@Incubating
+public interface PluginId {
```

Comment:
And to your question whether we need this: I think it is useful and we might actually change `AppliedPlugin` to have a `PluginId id()` method in the future and remove the duplicated methods over there.